ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -850,13 +850,32 @@ int ompi_osc_find_attached_region_position(ompi_osc_dynamic_win_info_t *dynamic_
                                                       base, len, insert);
     }
 }
+bool ompi_osc_need_acc_lock(ompi_osc_ucx_module_t *module, int target)
+{
+    ompi_osc_ucx_lock_t *lock = NULL;
+    opal_hash_table_get_value_uint32(&module->outstanding_locks,
+                                     (uint32_t) target, (void **) &lock);
+
+    /* if there is an exclusive lock there is no need to acqurie the accumulate lock */
+    return !(NULL != lock && LOCK_EXCLUSIVE == lock->type);
+}
+
 
 int ompi_osc_ucx_win_attach(struct ompi_win_t *win, void *base, size_t len) {
     ompi_osc_ucx_module_t *module = (ompi_osc_ucx_module_t*) win->w_osc_module;
     int insert_index = -1, contain_index;
     int ret = OMPI_SUCCESS;
 
+    bool lock_needed = ompi_osc_need_acc_lock(module, ompi_comm_rank(module->comm));
+    if (lock_needed) {
+        ompi_osc_ucx_lock(MPI_LOCK_EXCLUSIVE, ompi_comm_rank(module->comm), 0, win);
+    }
+
     if (module->state.dynamic_win_count >= OMPI_OSC_UCX_ATTACH_MAX) {
+        OSC_UCX_ERROR("Dynamic window attach failed: Cannot satisfy %d attached windows. "
+                "Max attached windows is %d \n",
+                module->state.dynamic_win_count+1,
+                OMPI_OSC_UCX_ATTACH_MAX);
```

Comment:
Missing unlock here and here: https://github.com/open-mpi/ompi/pull/10593/files#diff-c74691074f3004a065db3878842f74f3d8bfc7709cd1b4f1c42e8e2e4d1b66ddR914

Reply:
Alright, I'm going to move this check before the lock then.